kube-core/src/metadata.rs

```diff
@@ -17,6 +17,42 @@ pub struct TypeMeta {
     pub kind: String,
 }
 
+impl TypeMeta {
+    /// Construct a new `TypeMeta` for the object list from the given resource.
+    ///
+    /// ```
+    /// # use k8s_openapi::api::core::v1::Pod;
+    /// # use kube_core::TypeMeta;
+    ///
+    /// let type_meta = TypeMeta::list_kind::<Pod>();
+    /// assert_eq!(type_meta.kind, "PodList");
+    /// assert_eq!(type_meta.api_version, "v1");
+    /// ```
+    pub fn list_kind<K: Resource<DynamicType = ()>>() -> Self {
+        TypeMeta {
+            api_version: K::api_version(&()).into(),
+            kind: K::kind(&()).to_string() + "List",
+        }
+    }
+
+    /// Construct a new `TypeMeta` for the object from the given resource.
+    ///
+    /// ```
+    /// # use k8s_openapi::api::core::v1::Pod;
+    /// # use kube_core::TypeMeta;
+    ///
+    /// let type_meta = TypeMeta::kind::<Pod>();
+    /// assert_eq!(type_meta.kind, "Pod");
+    /// assert_eq!(type_meta.api_version, "v1");
+    /// ```
+    pub fn kind<K: Resource<DynamicType = ()>>() -> Self {
```

Comment:
hm, this brings up a minor point here about naming; `TypeMeta::kind` does sound like a setter for `typemeta.kind` which it currently isn't.

How about we name the two ctors:

- `TypeMeta::resource` -> for this one here that does kind based on resource only
- `TypeMeta::list` -> for the one that suffixes `List`